src/main/java/parsers/agmart/AgmartParser.java

```diff
@@ -0,0 +1,87 @@
+package parsers.agmart;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import parsers.CommodityPriceSource;
+import parsers.CrawlCommodityPriceDto;
+import parsers.ICommodityPricingParser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+
+public class AgmartParser implements ICommodityPricingParser {
+    private static final String agmartDateFormat = "yyy y-MM-dd";
+
+    public static void setDate(long date) {
+        AgmartParser.date = date;
+    }
+
+    private static long date;
+    private static HashMap<String,String>city_state=new HashMap<String,String>();
+    private static HashMap<String,String>id_state=new HashMap<String,String>();
```

Comment:
In case of multithreading (where more than one instance of crawling is running) it will create issue , make this variable method level

Reply:
But I declared them static so they will be only instance of the variable